src/index.js

```diff
@@ -30,6 +38,15 @@ class ErrorBoundary extends React.Component {
     const {fallbackRender, FallbackComponent, fallback} = this.props
 
     if (error !== null) {
+      // we'll get a re-render with the error state in getDerivedStateFromError
+      // but we don't have the info yet, so just render null
+      // note that this won't be committed to the DOM thanks to our componentDidCatch
+      // so the user won't see a flash of nothing, so this works fine.
+      // the benefit of doing things this way rather than just putting both the
+      // error and info setState within componentDidCatch is we avoid re-rendering
+      // busted stuff: https://github.com/bvaughn/react-error-boundary/issues/66
```

Comment:
This doesn't seem ideal, since we'll still double render (and cause two DOM mutations). My guess is that– for most cases– the component stack won't get rendered anyway. (Heck even the error text itself probably won't be rendered, (since you probably wouldn't want to show those to the end user).

You could always log them to the console in DEV mode using the `onError` prop if you wanted to make it faster to track down errors while iterating on code.

I wonder if we should reconsider this part of the API?

Reply:
Yeah, I don't like this either, but I'm not sure I see much of a drawback because really all this means is that there will be an uncommitted render of null which I don't think is a big deal. Unless I'm missing something.

Reply:
I'm definitely willing to make a major version bump to remove the componentStack.

Reply:
> all this means is that there will be an uncommitted render of null

Just to clarify, this would get "committed" (meaning the DOM would get updated) *before* `componentDidCatch` gets called– which would trigger another render+commit loop (which would update the DOM again).

It's not the end of the world, since error boundaries aren't expected to trigger in a tight loop like e.g. list items during scrolling, but it's not ideal to have the intermediate null. Seems like we could tighten it up a bit, if we were being picky 😁

Reply:
Ah, yeah, spices like I was misunderstanding things then. I'll make the breaking change this requires. Thanks!